### PR TITLE
Refactoring/5240 single status annotation

### PR DESCRIPTION
**Pull request by melsawy**

### CODE REVIEW

File: src/app/relay/DeleteStatusMutation.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  getFatQuery() {
+    let query = '';
+    switch (this.props.parent_type) {
+    case 'source':
+      query = Relay.QL`fragment on DestroyStatusPayload { deletedId, source { annotations, tags } }`;
+      break;
+    case 'project_media':
+      query = Relay.QL`fragment on DestroyStatusPayload { deletedId, project_media { annotations, tags, last_status, last_status_obj { id } } }`;
```
Comment (caiosba) [inline src/app/relay/DeleteStatusMutation.js]: You can remove `tags`

File: src/app/relay/DeleteStatusMutation.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  getVariables() {
+    return { id: this.props.id };
+  }
+
+  getFatQuery() {
+    let query = '';
+    switch (this.props.parent_type) {
+    case 'source':
+      query = Relay.QL`fragment on DestroyStatusPayload { deletedId, source { annotations, tags } }`;
```
Comment (caiosba) [inline src/app/relay/DeleteStatusMutation.js]: You can remove `tags`

File: src/app/relay/DeleteStatusMutation.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    fieldIds[this.props.parent_type] = this.props.annotated.id;
+
+    return [
+      {
+        type: 'NODE_DELETE',
+        parentName: this.props.parent_type,
+        parentID: this.props.annotated.id,
+        connectionName: 'tags',
+        deletedIDFieldName: 'deletedId',
+      },
```
Comment (caiosba) [inline src/app/relay/DeleteStatusMutation.js]: You can remove this whole block that starts on line 37.

File: src/app/relay/DeleteVersionMutation.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        deletedIDFieldName: 'deletedId',
+      },
+      {
+        type: 'FIELDS_CHANGE',
+        fieldIDs: fieldIds,
+      },
+    ];
+  }
+}
+
```
Comment (caiosba) [inline src/app/relay/DeleteVersionMutation.js]: Same comments as `DeleteStatusMutation.js`

File: src/app/relay/UpdateStatusMutation.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+ getFatQuery() {
+    let query = '';
+    switch (this.props.parent_type) {
+    case 'source':
+      query = Relay.QL`fragment on UpdateStatusPayload { statusEdge, source { annotations, id } }`;
+      break;
+    case 'project_media':
+      query = Relay.QL`fragment on UpdateStatusPayload { statusEdge, project_media { annotations, id, last_status, annotations_count } }`;
```
Comment (caiosba) [inline src/app/relay/UpdateStatusMutation.js]: No need to add `last_status_obj` here?

File: src/app/relay/UpdateStatusMutation.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        name: this.props.annotator.name,
+        profile_image: this.props.annotator.profile_image,
+      },
+      medias: {
+        edges: [],
+      },
+    };
+
+    const media = Object.assign({}, this.props.annotated);
+    media.last_status = this.props.annotation.status;
```
Comment (caiosba) [inline src/app/relay/UpdateStatusMutation.js]: No need to add `last_status_obj` here?

Comment (melsawy) [inline src/app/relay/UpdateStatusMutation.js]: yes, no need to add last_status_obj

File: src/app/relay/UpdateStatusMutation.js (hunk 5 above)
Comment (melsawy) [inline src/app/relay/UpdateStatusMutation.js]: yes

File: src/app/relay/DeleteVersionMutation.js (hunk 4 above)
Comment (melsawy) [inline src/app/relay/DeleteVersionMutation.js]: Done

File: src/app/relay/DeleteStatusMutation.js (hunk 3 above)
Comment (melsawy) [inline src/app/relay/DeleteStatusMutation.js]: Done

File: src/app/relay/DeleteStatusMutation.js (hunk 2 above)
Comment (melsawy) [inline src/app/relay/DeleteStatusMutation.js]: Done

File: src/app/relay/DeleteStatusMutation.js (hunk 1 above)
Comment (melsawy) [inline src/app/relay/DeleteStatusMutation.js]: Done

**Pull request merged**